addons/binding/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/handler/TrackerHandler.java

```diff
@@ -0,0 +1,356 @@
+            Channel systemDistance = thing.getChannel(CHANNEL_DISTANCE_SYSTEM_ID);
+            ChannelBuilder channelBuilder = null;
+            if (systemDistance != null) {
+                if (!systemDistance.getConfiguration().get(CONFIG_REGION_CENTER_LOCATION).equals(sysLocation.toFullString())) {
+                    logger.trace("Existing distance channel for system. Changing system location config parameter: {}", sysLocation.toFullString());
+
+                    channelBuilder = callback.editChannel(thing, systemDistanceChannelUID);
+                    Configuration configToUpdate = systemDistance.getConfiguration();
+                    configToUpdate.put(CONFIG_REGION_CENTER_LOCATION, sysLocation.toFullString());
+                    channelBuilder.withConfiguration(configToUpdate);
+                } else {
+                    logger.trace("Existing distance channel for system. No change.");
+                }
+            } else {
+                logger.trace("Creating missing distance channel for system.");
+
+                Configuration config = new Configuration();
+                config.put(ConfigHelper.CONFIG_REGION_NAME, CHANNEL_DISTANCE_SYSTEM_NAME);
+                config.put(CONFIG_REGION_CENTER_LOCATION, sysLocation.toFullString());
+                config.put(ConfigHelper.CONFIG_REGION_RADIUS, CHANNEL_DISTANCE_SYSTEM_RADIUS);
+
+                channelBuilder = callback.createChannelBuilder(systemDistanceChannelUID, CHANNEL_TYPE_DISTANCE)
+                        .withLabel("System Distance")
+                        .withConfiguration(config);
+            }
+
+            //update the thing with system distance channel
+            if (channelBuilder != null) {
+                List<Channel> channels = new ArrayList<>(thing.getChannels());
+                if (systemDistance != null) {
+                    channels.remove(systemDistance);
+                }
+                channels.add(channelBuilder.build());
+
+                ThingBuilder thingBuilder = editThing();
+                thingBuilder.withChannels(channels);
+                updateThing(thingBuilder.build());
+
+                logger.debug("Distance channel created for system: {}", systemDistanceChannelUID);
+            }
+        }
+    }
+
+    /**
+     * Create a map of all configured distance channels to handle channel updates easily.
+     */
+    private void mapDistanceChannels() {
+        distanceChannelMap = thing.getChannels().stream()
+                .filter(c -> CHANNEL_TYPE_DISTANCE.equals(c.getChannelTypeUID()))
+                .collect(Collectors.toMap(c -> ConfigHelper.getRegionName(c.getConfiguration()), Function.identity()));
+        //register the collected regions
+        regions.addAll(distanceChannelMap.keySet());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        //no command handling
```

Comment:
Ok, I stumbled over this during my testing. The REFRESH command should be handled whenever possible.
Clearly, you cannot request a more recent location or battery status from the tracker, but you should send at least the last known value as a status update. Especially the distance channels can and should be refreshed, if a REFRESH command is sent to them. I noticed that, because after adding a distance channel, its state stayed -NaN, although a location was available. If it correctly responds to REFRESH commands, this should work automatically.

Reply:
I'll preserve the last received message from the tracker and replay it when REFRESH is needed. This should solve this issue.